packages/@aws-cdk/aws-amplify/lib/app.ts

```diff
@@ -0,0 +1,347 @@
+  /**
+   * Allow creating from App ID
+   *
+   * @param scope
+   * @param id
+   * @param appId
+   *
+   * @returns Stubbed App
+   */
+  public static fromAppId(scope: Construct, id: string, appId: string): IApp {
+    class Import extends Resource implements IApp {
+      public readonly appId = appId;
+    }
+
+    return new Import(scope, id);
+  }
+
+  /**
+   * Unique Id for the Amplify App.
+   *
+   * @attribute
+   */
+  public readonly appId: string;
+
+  /**
+   * Name for the Amplify App.
+   *
+   * @attribute
+   */
+  public readonly appName: string;
+
+  /**
+   * ARN for the Amplify App.
+   *
+   * @attribute
+   */
+  public readonly appArn: string;
+
+  /**
+   * Default domain for the Amplify App.
+   *
+   * @attribute
+   */
+  public readonly appDefaultDomain: string;
+
+  /**
+   * Resource
+   */
+  private readonly resource: CfnApp;
+
+  private readonly basicAuthResolver: BasicAuthResolver = new BasicAuthResolver();
+
+  private readonly environmentVariablesResolver: EnvironmentVariablesResolver = new EnvironmentVariablesResolver();
+
+  private readonly customRulesResolver: CustomRulesResolver = new CustomRulesResolver();
+
+  constructor(scope: Construct, id: string, props: AppProps) {
+    super(scope, id, {
+      physicalName: props.appName
+    });
+
+    if (props.basicAuthConfig) {
+      this.addBasicAuth(props.basicAuthConfig.username, props.basicAuthConfig.password);
+    }
+
+    if (props.environmentVariables && props.environmentVariables.length > 0) {
+      this.environmentVariablesResolver.addEnvironmentVariables(...props.environmentVariables);
+    }
+
+    let roleArn: string | undefined;
+
+    if (props.serviceRole) {
+      roleArn = props.serviceRole.roleArn;
+    }
+
+    if (props.buildSpec && !props.serviceRole) {
+      const role = new LazyRole(scope, 'lazy-role', {
+        assumedBy: new ServicePrincipal('amplify.amazonaws.com')
+      });
+
+      roleArn = role.roleArn;
+    }
+
+    const resource = new CfnApp(this, 'App', {
+      accessToken: props.accessToken,
+      basicAuthConfig: this.basicAuthResolver,
+      buildSpec: (props.buildSpec) ? props.buildSpec.toBuildSpec() : undefined,
+      customRules: this.customRulesResolver,
+      description: props.description,
+      environmentVariables: this.environmentVariablesResolver,
+      iamServiceRole: roleArn,
+      name: props.appName,
+      oauthToken: (props.oauthToken && props.oauthToken.toString()) ? props.oauthToken.toString() : undefined,
+      repository: props.repository,
+      tags: props.tags
+    });
+
+    this.appArn = resource.attrArn,
+    this.appDefaultDomain = resource.attrDefaultDomain;
+    this.appId = resource.attrAppId;
+    this.appName = resource.attrAppName;
+
+    this.resource = resource;
+  }
+
+  /**
+   * Add Service Role
+   *
+   * @param role
+   */
+  public addServiceRole(role: IRole) {
+    this.resource.iamServiceRole = role.roleArn;
```

Comment:
Do we really need this mutator? Seems like there could be some conflicts with the fact that you are defining a LazyRole above... If there is no clear use case to be able to mutate configuration, we try to avoid it and only allow people to configure during initialization.